core/src/main/java/org/apache/druid/java/util/http/client/response/SequenceInputStreamResponseHandler.java

```diff
@@ -102,14 +100,14 @@ public InputStream nextElement()
   @Override
   public ClientResponse<InputStream> handleChunk(
       ClientResponse<InputStream> clientResponse,
-      HttpChunk chunk,
+      HttpContent chunk,
       long chunkNum
   )
   {
-    final ChannelBuffer channelBuffer = chunk.getContent();
-    final int bytes = channelBuffer.readableBytes();
+    final ByteBuf byteBuf = chunk.content();
+    final int bytes = byteBuf.readableBytes();
     if (bytes > 0) {
-      try (ChannelBufferInputStream channelStream = new ChannelBufferInputStream(channelBuffer)) {
+      try (ByteBufInputStream channelStream = new ByteBufInputStream(byteBuf)) {
```

Comment:
@gianm I think this try-with-resources is incorrect, we don't want to be closing the stream at the end of the block, but instead let the downstream SequenceInputStream close the stream when done reading it.

Reply:
It looks like ChannelBufferInputStream.close does nothing, so it didn't used to matter. But yeah, this code doesn't look right. It hands off the input stream and then closes it, which is poor form.